syntax = "proto3";

package influence_service_copy;

service InfluenceServiceCopy {
    rpc ComputeInfluences (stream InfluenceRequest) returns (InfluenceResponse);
    rpc ComputePDP (PDPRequest) returns (PDPResponse);
    rpc ComputePDP2D (PDP2DRequest) returns (PDP2DResponse);
    rpc ComputeALE (ALERequest) returns (ALEResponse);
} 

message InfluenceRequest {
    bytes train_data = 1;
    bytes model = 2;
    bytes test_data = 3;
    bytes train_labels = 4;
    bytes test_labels = 5;
    int32 num_influential = 6;
    bytes untransformed_train = 7;
    bytes untransformed_train_labels = 8;
}

message InfluenceResponse {
    repeated float influences = 1;
    bytes positive = 2;
    bytes negative = 3;
}

message PDPRequest{
    string param_grid = 1;
    bytes model = 2;
}

message PDPResponse{
    string hp_values = 1;
    string pdp_values = 2;
}

message PDP2DRequest{
    string param_grid = 1;
    bytes model = 2;
    string feature1 = 3;
    string feature2 = 4;
}

message PDP2DResponse{
    string xi = 1;
    string yi = 2;
    string zi = 3;
}

message ALERequest{
    bytes data = 1;
    string param_grid = 2;
    bytes model = 3;
}

message ALEResponse{
    string data = 1;
}